grades/api.py

```diff
@@ -0,0 +1,155 @@
+"""
+APIs for the grades app
+"""
+import logging
+from collections import namedtuple
+
+from django.contrib.auth.models import User
+
+from dashboard.api_edx_cache import CachedEdxUserData, CachedEdxDataApi
+from dashboard.models import CachedEnrollment
+from grades.models import (
+    FinalGrade,
+    FinalGradeStatus,
+)
+
+
+log = logging.getLogger(__name__)
+
+UserFinalGrade = namedtuple('UserFinalGrade', ['grade', 'passed'])
+
+
+def _compute_grade_for_fa(user_edx_run_data):
+    """
+    Gets the final grade for an user enrolled in the course run that is part of a financial aid program.
+
+    Args:
+        user_edx_run_data (dashboard.api_edx_cache.UserCachedRunData): the edx cached data for an user in a course run
+
+    Returns:
+        UserFinalGrade: a namedtuple of (float, bool,) representing the final grade
+            of the user in the course run and whether she passed it
+    """
+    run_passed = None
+    grade = None
+    if user_edx_run_data.certificate is not None:
+        run_passed = True
+        grade = user_edx_run_data.certificate.grade
+    else:
+        run_passed = user_edx_run_data.current_grade.passed
+        grade = user_edx_run_data.current_grade.percent
+    return UserFinalGrade(grade=grade, passed=run_passed)
+
+
+def _compute_grade_for_non_fa(user_edx_run_data):
+    """
+    Gets the final grade for an user enrolled in the
+    course run that is part of a non financial aid program.
+    If the user has a certificate, she has passed otherwise she has not.
+
+    Args:
+        user_edx_run_data (dashboard.api_edx_cache.UserCachedRunData): the edx cached data for an user in a course run
+
+    Returns:
+        UserFinalGrade: a namedtuple of (float, bool,) representing the final grade
+            of the user in the course run and whether she passed it
+    """
+    run_passed = user_edx_run_data.certificate is not None
```

Comment:
It should be verified in this case. 
I am tempted to put a check about it, but the reality is that the existence of a certificate for a non_fa course should mean that it is verified (meaning there should never be a case where there are both `verified` and `honor` certificates for non_fa